### PR TITLE
Add functionality to verify passes

### PR DESCRIPTION
**Pull request by aditiii23**

### CODE REVIEW

File: frontend/src/components/Profile/Profile.js [hunk 1]
```diff
@@ -34,6 +36,18 @@ const Profile = () => {
...
+          className={profilestyle.button}
+          onClick={() => {
+            setgenerate(true)
+            navigate("/viewUnverifiedPass")
+          }}
+        >
+        Verify Pass
+        </button>
+      ) : null}
+      {verify ? <VerifyPass /> : null}
```
Comment (mrpriyansh) [inline frontend/src/components/Profile/Profile.js]: Add this with the above condition.

File: frontend/src/components/VerifyPass/VerifyPass.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      <div className={verifyPassesStyle.item}>
+        <p> {item.name}</p>
+        <p> {item.email}</p>
+        <p>{item.phone}</p>
+        <p>{item.duration}</p>
+        <p>{startDate?.substring(0, 10)}</p>
+        <p>{startDate?.substring(11, startDate.length - 8)}</p>
+        <Link
+          className={verifyPassesStyle.button_common}
+          onClick={async () => {
```
Comment (mrpriyansh) [inline frontend/src/components/VerifyPass/VerifyPass.js]: Declare the onClick function somewhere in the component and use it here.

File: frontend/src/components/VerifyPass/ViewUnverifiedPass.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from "react"
+import verifyPassesStyle from "./VerifyPass.module.css"
+import axios from "axios"
+import Verify from "./VerifyPass"
```
Comment (mrpriyansh) [inline frontend/src/components/VerifyPass/ViewUnverifiedPass.js]: What is Verify?

File: frontend/src/components/VerifyPass/VerifyPass.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import * as React from "react"
```
Comment (mrpriyansh) [inline frontend/src/components/VerifyPass/VerifyPass.js]: Isn't this similar to Pass.js

File: frontend/src/components/VerifyPass/ViewUnverifiedPass.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from "react"
```
Comment (mrpriyansh) [inline frontend/src/components/VerifyPass/ViewUnverifiedPass.js]: Isn't this similar to ViewPasses.js? If so, please try to use some conditional statements and utilize the same component.

**Review by mrpriyansh — CHANGES_REQUESTED** [review]

File: frontend/src/App.js [hunk 6]
```diff
@@ -34,7 +34,14 @@ const Routing = () => {
         <Route path="/register" element={<Signup />}></Route>
         <Route path="/" element={<Profile />}></Route>
         <Route path="/generatePass" element={<GeneratePass />}></Route>
-        <Route path="/viewPasses" element={<ViewPasses />}></Route>
+        <Route
+          path="/viewPasses"
+          element={<ViewPasses verify={false} abc={"1234"} />}
```
Comment (mrpriyansh) [inline frontend/src/App.js]: Remove abcd

Comment (mrpriyansh) [reply frontend/src/App.js]: Try to remove verify and observe the changes. And check how it impacts

File: frontend/src/components/Profile/Profile.js [hunk 7]
```diff
@@ -1,39 +1,38 @@
-import React, { useState } from "react"
+import React from "react"
 import profilestyle from "./Profile.module.css"
-import GeneratePass from "../GeneratePass/GeneratePass"
-import ViewPasses from "../ViewPasses/ViewPasses"
 import { useNavigate } from "react-router-dom"
 
+//check for role for button display (array element) user, guard, admin
```
Comment (mrpriyansh) [inline frontend/src/components/Profile/Profile.js]: Remove comments

File: frontend/src/components/ViewPasses/Pass.js [hunk 8]
```diff
@@ -1,27 +1,50 @@
...
+import axios from "axios"
+import { apiUrl } from "../../services/config"
+import { Link, useNavigate } from "react-router-dom"
 
-export default function Pass(item) {
+export default function Pass({item, verify}) {
+  const navigate = useNavigate()
   const startDate = item.start
+  const verifyPass = async () => {
+    await axios.put(
```
Comment (mrpriyansh) [inline frontend/src/components/ViewPasses/Pass.js:11]: Check the standard way to use put(specifically regarding payload)

File: frontend/src/components/ViewPasses/Pass.js [hunk 9]
```diff
@@ -1,27 +1,50 @@
...
+        <p>{item.generateId}</p>
+        <p>{item.name}</p>
+        <p>{item.email}</p>
         <p>{item.phone}</p>
         <p>{item.duration}</p>
-        <p>{startDate?.substring(0,10)}</p>
-        <p>{startDate?.substring(11, startDate.length-8)}</p>
+        <p>{startDate?.substring(0, 10)}</p>
+        <p>{startDate?.substring(11, startDate.length - 8)}</p>
+        {verify == true ? (
```
Comment (mrpriyansh) [inline frontend/src/components/ViewPasses/Pass.js:41]: What is the type of verify?

File: frontend/src/components/ViewPasses/ViewPasses.js [hunk 10]
```diff
@@ -5,7 +5,7 @@ import Pass from "./Pass"
 import { apiUrl } from "../../services/config"
 import { toast } from "react-toastify"
 
-const ViewPasses = () => {
+const ViewPasses = (props) => {
```
Comment (mrpriyansh) [inline frontend/src/components/ViewPasses/ViewPasses.js]: Destructor props

**Review by mrpriyansh — CHANGES_REQUESTED** [review]

File: frontend/src/components/Profile/Profile.js (hunk 7 above)
Comment (aditiii23) [reply frontend/src/components/Profile/Profile.js]: sorry i forgot to remove them, removed now.

File: frontend/src/App.js (hunk 6 above)
Comment (aditiii23) [reply frontend/src/App.js]: When I removed verify prop from viewPasses, the if-else statements in the useEffect only passed the else part, so every time viewUnverifiedPasses is called, viewPasses will never be called..

File: frontend/src/components/ViewPasses/Pass.js (hunk 9 above)
Comment (aditiii23) [reply frontend/src/components/ViewPasses/Pass.js:41]: verify is an object

File: frontend/src/components/ViewPasses/ViewPasses.js (hunk 10 above)
Comment (aditiii23) [reply frontend/src/components/ViewPasses/ViewPasses.js]: done

Comment (aditiii23) [reply frontend/src/components/ViewPasses/ViewPasses.js]: only verify needed to be destructured in this one

File: frontend/src/App.js [hunk 11]
```diff
@@ -34,7 +34,14 @@ const Routing = () => {
         <Route path="/register" element={<Signup />}></Route>
         <Route path="/" element={<Profile />}></Route>
         <Route path="/generatePass" element={<GeneratePass />}></Route>
-        <Route path="/viewPasses" element={<ViewPasses />}></Route>
+        <Route
+          path="/viewPasses"
+          element={<ViewPasses verify={false} />}
```
Comment (mrpriyansh) [inline frontend/src/App.js]: Did you checked without verify={false} ?

File: frontend/src/components/ViewPasses/Pass.js (hunk 9 above)
Comment (mrpriyansh) [reply frontend/src/components/ViewPasses/Pass.js:41]: sure?

**Review by mrpriyansh — CHANGES_REQUESTED** [review]

File: frontend/src/App.js (hunk 11 above)
Comment (aditiii23) [reply frontend/src/App.js]: yes

Comment (aditiii23) [reply frontend/src/App.js]: firstly I did it without using verify={false} then only the use Effect's else part was getting executed, then I had put this condition

**Review by mrpriyansh — APPROVED** [review]

**Pull request merged**